solutions/swb-app/src/userRoutes.ts

```diff
@@ -54,15 +55,38 @@ export function setUpUserRoutes(router: Router, user: UserManagementService): vo
         await user.deleteUser(userId);
         res.status(204).send();
       } catch (err) {
-        if (err instanceof UserNotFoundError) {
-          throw Boom.notFound(`Could not find user ${userId}`);
-        }
+        if (err instanceof UserNotFoundError) throw Boom.notFound(`Could not find user ${userId}`);
+        if (Boom.isBoom(err)) throw err;
+        throw Boom.badImplementation(`Could not delete user ${userId}`);
+      }
+    })
+  );
 
-        if (Boom.isBoom(err)) {
-          throw err;
+  router.patch(
+    '/users/:userId',
+    wrapAsync(async (req: Request, res: Response) => {
+      processValidatorResult(validate(req.body, UpdateUserSchema));
+      const userId = req.params.userId;
+      try {
+        const existingUser = await user.getUser(userId);
+
+        if (!_.isUndefined(req.body.status)) {
+          if (req.body.status === 'ACTIVE' && existingUser.status === Status.INACTIVE)
+            await user.activateUser(userId);
+          if (req.body.status === 'INACTIVE' && existingUser.status === Status.ACTIVE)
+            await user.deactivateUser(userId);
+          delete req.body.status; // Status update is complete
         }
 
-        throw Boom.badImplementation(`Could not delete user ${userId}`);
+        // Since updateUser() requires object of type User
+        const userToUpdate = _.assign(existingUser, req.body);
+
+        await user.updateUser(userId, userToUpdate);
+        res.status(204).send();
+      } catch (err) {
```

Comment:
`user.updateUser` also throws `InvalidParameterError` if the email is not a valid email or its already in use.
`user.addUserToRole` will throw `RoleNotFoundError` if the group doesnt exist in the user pool